### PR TITLE
Features Guide v1

### LINKED ISSUE #9461: Add Enabling/Disabling Features guide

### PR DESCRIPTION
**Pull request by DGuhr**
Initial Version for Features guide

Closes #9461

### CODE REVIEW

File: docs/guides/src/main/server/features.adoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+<@tmpl.guide
+title="Enabling and disabling features"
+summary="Understand how to configure Keycloak to use optional features"
+includedOptions="features features.*">
+
+Keycloak has packed some functionality in features, some of them not enabled by default. These features include features that are in tech preview or deprecated features. In addition there are some features that are enabled by default, but can be disabled if you don't need them for your specific usage scenario.
+
+== List of optional features
+
+[cols="4*", options="header"]
```
Comment (DGuhr) [inline docs/guides/src/main/server/features.adoc:14]: @stianst fyi: this is why i created https://github.com/keycloak/keycloak/issues/9549 - also the dependency I had to add and the change in Options.java should not be necessary anymore when #9549 is solved.

File: docs/maven-plugin/src/main/java/org/keycloak/guides/maven/Options.java [hunk 2]
```diff
@@ -18,6 +21,21 @@ public Options() {
                 .filter(m -> !m.isHidden())
                 .map(m -> new Option(m.getFrom(), m.getCategory(), m.isBuildTime(), m.getDescription(), m.getDefaultValue(), m.getExpectedValues()))
                 .collect(Collectors.toMap(Option::getKey, o -> o, (o1, o2) -> o1)); // Need to ignore duplicate keys??
+
+        addFeatureOptions(options);
```
Comment (DGuhr) [inline docs/maven-plugin/src/main/java/org/keycloak/guides/maven/Options.java:25]: see comment above.

File: docs/guides/src/main/server/features.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+|Yes
+|Preview
+|===
+
+=== Enabling/Disabling features
+
+Features need to be enabled/disabled using the `build` command.
+To enabe a specific feature, for example to enable the ability to use openshift integration `openshift_integration`, run:
+
+<@kc.build parameters="--features-openshift_integration=enabled"/>
```
Comment (pedroigor) [inline docs/guides/src/main/server/features.adoc:97]: I'm not sure if `openshift` feature is the best example. Does it make more sense to use a more common feature like `token_exchange`, for instance?

The main point is to only mention what we want people to use.

**Comment by stianst** [comment]
Updated this PR to use Profile as the source of the list of features, combined the two tables, and listing different types of features under different headings:

https://github.com/keycloak/keycloak/pull/9604

**Comment by DGuhr** [comment]
Closing this one, then ;)

**Pull request closed without merging**